pkg/controller/replica.go

```diff
@@ -171,6 +175,37 @@ func (r *ReconcileReplica) triggerGarbageCollection(volumeName string) {
 			)
 		}
 	}(deletionCtx)
+
+	go func(ctx context.Context) {
+		err := wait.PollImmediate(deletionPollingInterval, 10*time.Minute, func() (bool, error) {
+			volRequirement, err := azureutils.CreateLabelRequirements(consts.VolumeNameLabel, selection.Equals, volumeName)
+			if err != nil {
+				return false, err
+			}
+			labelSelector := labels.NewSelector().Add(*volRequirement)
+			azVolumeAttachmentList := &diskv1beta1.AzVolumeAttachmentList{}
+			err = r.controllerSharedState.cachedClient.List(ctx, azVolumeAttachmentList, &client.ListOptions{LabelSelector: labelSelector})
+			if err != nil {
+				if apiErrors.IsNotFound(err) {
+					return true, nil
+				}
+				klog.Errorf("failed to get AzVolumeAttachments: %v", err)
+				return false, err
+			}
+			if len(azVolumeAttachmentList.Items) == 0 {
+				return true, nil
+			}
+			return false, nil
```

Comment:
np: we can probably just do `return len(azVolumeAttachmentList.Items) == 0, nil` instead of having a separate if clause for this.